### DISCUSSION TITLE
What is the recommended way to do something as soon as all the props are loaded ?

### DISCUSSION THREAD

**Discussion by 0xbuidler** [lit/lit#4606] [General]
The title is pretty explicit but basically I want to execute a function when all the props of my LitElement are loaded, how can I do that ?

**Comment by justinfagnani** [comment]
There really isn't a concept like "all the props are loaded", because properties on the element (or other input that's not properties, like children) could change at any moment.

Consider creating an element like:
```ts
const el = document.createElement('my-element');
el.foo = 'foo';
el.bar = 'bar';
setTimeout(() => el.foo = 'baz', 1000);
```

The best we can say is that a property has been set, and then we batch changes so that you can be pretty sure that a set of synchronous changes (like construction, setting foo and setting bar as above) have happened.

If you really want to go after the first batch, you can implement [`firstUpdated()`](https://lit.dev/docs/components/lifecycle/#firstupdated), although I'd recommend that this is mainly used for getting DOM that's rendered during the first render. Otherwise I'd say to try to design the element so that you respond to any property changes an implement logic in `willUpdate()` or `updated()` as needed.
